Remote agent pprof endpoints (#6841)
hashicorp/nomad: website/source/api/agent.html.md

by ACLs and supports remote forwarding to a client node or server. See the
[Golang documentation](https://golang.org/pkg/runtime/pprof/#Profile) for a list of available profiles.

| Method | Path                          | Produces                   |
| ------ | ----------------------------- | -------------------------- |
| `GET`  | `/agent/pprof/cmdline`        | `text/plain`               |
| `GET`  | `/agent/pprof/profile`        | `application/octet-stream` |
| `GET`  | `/agent/pprof/trace`          | `application/octet-stream` |
| `GET`  | `/agent/pprof/<pprof profile>`| `application/octet-stream` |

The table below shows this endpoint's support for
[blocking queries](/api/index.html#blocking-queries) and
[required ACLs](/api/index.html#acls).

| Blocking Queries | ACL Required  |
| ---------------- | ------------- |
| `NO`             | `agent:write` |

### Default Behavior

The default behavior of this endpoint complex but seeks to maximize security,
backward compatibility, and still allow debug access by default when possible. 
The table below outlines the different scenarios which will enable or disable

> the endpoint.

Reviewer: Your text isn't bad, but here's another stab at it. I'm ok with shipping either.

> This endpoint is enabled whenever ACLs are enabled unless [`enable_debug`][enable_debug] is explicitly disabled. Due to the potentially sensitive nature of data contained in profiles as well as their significant performance impact, the agent/pprof endpoint is protected by a high level ACL: `agent:write`. For these reasons it is recommended to leave `enable_debug` unset and only use the ACL-protected endpoints.
>
> The following table explains when each endpoint is available:

|    Endpoint      |  `enable_debug`  |  ACLs  |  **Available?**  |
|------------------|------------------|--------|------------------|
| /v1/agent/pprof  |  unset           |  n/a   |  no              |
| /v1/agent/pprof  |  `true`          |  n/a   |  yes             |
| /v1/agent/pprof  |  `false`         |  n/a   |  no              |
| /v1/agent/pprof  |  unset           |  off   |  no              |
| /v1/agent/pprof  |  unset           |  on    |  **yes**         |
| /v1/agent/pprof  |  `true`          |  off   |  yes             |
| /v1/agent/pprof  |  `false`         |  n/a   |  **no**          |

### Parameters

- `node_id` `(string: "a57b2adb-1a30-2dda-8df0-25abb0881952")` - Specifies a text

>   string containing a node-id to target for streaming.

Reviewer: ```suggestion
  string containing a Node ID to target for streaming.
```

- `server_id` `(string: "server1.global")` - Specifies a text

>   string containing a server id, name or "leader" to target a specific remote

Reviewer: ```suggestion
  string containing a server ID, name, or `leader` to target a specific remote
```

  server or leader for streaming.

- `seconds` `(int: 3)` - Specifies the amount of time to run a profile or trace
  request for.

- `debug` `(int: 1)` - Specifies if a given pprof profile should be returned as

>   text/plain instead of application/octet-stream. Defaults to 0, setting to 1

Reviewer: I think we should focus on "human vs machine" readable and not what the content type is (if I'm understanding this flag right!).

```suggestion
  human readable plain text instead of the pprof binary format. Defaults to 0, setting to 1 enables the human readable format.
```

  enables.

### Sample Request

```text

> $ curl -O -J \

Reviewer: Let's include a fake ACL token header since that's going to be the common use case.
